File: Client_Root/Client/Assets/Scripts/PlayerInput/PlayerInputs/GameItem.cs
using UnityEngine;
using FlatBuffers;

namespace PlayerInput
{
    public class GameItem : IPlayerInput
    {
        public int m_nPlayerIndex;
        public int m_nGameItemID;

        public override FBS.PlayerInputType GetPlayerInputType()
        {
            return FBS.PlayerInputType.GameItem;
        }

        public override byte[] Serialize()
        {
            FBS.PlayerInput.GameItem.StartGameItem(m_Builder);
            FBS.PlayerInput.GameItem.AddPlayerIndex(m_Builder, m_nPlayerIndex);
            FBS.PlayerInput.GameItem.AddGameItemID(m_Builder, m_nGameItemID);
            var data = FBS.PlayerInput.GameItem.EndGameItem(m_Builder);

            m_Builder.Finish(data.Value);

            return m_Builder.SizedByteArray();
        }

        public override bool Deserialize(byte[] bytes)
        {
            var buf = new ByteBuffer(bytes);

            var data = FBS.PlayerInput.GameItem.GetRootAsGameItem(buf);

            m_nPlayerIndex = data.PlayerIndex;
            m_nGameItemID = data.GameItemID;

            return true;
        }

        public override string ToString()
        {
            return string.Format("PlayerIndex : {0}, GameItemID : {1}", m_nPlayerIndex, m_nGameItemID);
        }
    }
}

File: Server_Root/Server/RoomServer/PlayerInput/PlayerInputs/GameItem.cpp
#include "stdafx.h"
#include "GameItem.h"
#include "../../../FBSFiles/PlayerInput/GameItem_generated.h"

namespace PlayerInput
{
	GameItem::GameItem()
	{
	}

	GameItem::~GameItem()
	{
	}

	FBS::PlayerInputType GameItem::GetType()
	{
		return FBS::PlayerInputType::PlayerInputType_GameItem;
	}

	IPlayerInput* GameItem::Clone()
	{
		GameItem* pClone = new GameItem();

		pClone->m_nPlayerIndex = m_nPlayerIndex;
		pClone->m_nGameItemID = m_nGameItemID;

		return pClone;
	}

	const char* GameItem::Serialize(int* pLength)
	{
		FBS::PlayerInput::GameItemBuilder data_builder(m_Builder);
		data_builder.add_PlayerIndex(m_nPlayerIndex);
		data_builder.add_GameItemID(m_nGameItemID);
		auto data = data_builder.Finish();

		m_Builder.Finish(data);

		*pLength = m_Builder.GetSize();

		return (char*)m_Builder.GetBufferPointer();
	}

	bool GameItem::Deserialize(const char* pChar)
	{
		auto data = flatbuffers::GetRoot<FBS::PlayerInput::GameItem>((const void*)pChar);

		m_nPlayerIndex = data->PlayerIndex();
		m_nGameItemID = data->GameItemID();

		return true;
	}
}

